fem/lininteg.cpp

```diff
@@ -48,7 +48,7 @@ void DomainLFIntegrator::AssembleRHSElementVect(const FiniteElement &el,
       Tr.SetIntPoint (&ip);
       double val = Tr.Weight() * Q.Eval(Tr, ip);
 
-      el.CalcShape(ip, shape);
+      el.CalcPhysShape(Tr, shape);
```

Comment:
Is it correct that this change just divides by the weight for the `INTEGRAL` map type? Why is this change needed, and why did it not matter until now? Will this change results for systems that use the `INTEGRAL` type?

Reply:
Yes, this just devides by Det[J] for the `INTEGRAL` map type. Previously this integrator did not support map type `INTEGRAL` and using this combination would quietly produce the wrong answer. This is another small fix for the issue that Dan raised some time ago in #1037. It's my understanding that people generally don't use the `INTEGRAL` map type so I doubt anyone will notice. However, those who do choose to use it will find that this integrator works as expected and they won't need to track down the source of oddly scaled results.

Reply:
This can probably be optimized to avoid computing `Tr.Weight()` for `INTEGRAL` map-type -- we're now multiplying by that and then dividing by it. 😁

Reply:
@v-dobrev, you are correct of course. I prefer using `CalcPhysShape` because it hides details of the basis functions from the integrator which I find makes the integrator cleaner and easier to follow. For me this is worth the loss of this particular optimization. However, I'll be happy to change this if you think it is important to do so.

Reply:
I don't think it is critical to do the optimization, so you I'm okay to leave it as it is.